packages/office-ui-fabric-react/src/utilities/dateMath/DateMath.ts

```diff
@@ -85,6 +85,12 @@ export function setMonth(date: Date, month: number): Date {
  * @return {boolean} True if the two dates represent the same date (regardless of time-of-day), false otherwise.
  */
 export function compareDates(date1: Date, date2: Date): boolean {
+  if (date1 == null && date2 == null) {
```

Comment:
@MLoughry  `== null` actually also covers `undefined` (notice it's `==` not `===`).. but we might be more explicit :)